posts/2020-11-20-JNDI-gRPC-200012.adoc

```diff
@@ -0,0 +1,389 @@
+   </session>
+----
+
+Specify a component id (a prefix for default long form bindings)
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" component-id="Dept549/AccountProcessor"/>
+----
+
+Simple binding name (one name used for both local and remote)
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" simple-binding-name="ejb/AccountService"/>
+----
+
+Local and Remote home specific binding names
+
+
+[source, XML]
+----
+   <session name="AccountServiceBean" local-home-binding-name="ejblocal:AccountService"/>
+   <session name="AccountServiceBean" remote-home-binding-name="ejb/services/AccountService"/>
+----
+
+[customserver]
+==== Specify Custom bindings in server.xml
+
+Following is an example of how to configure custom bindings for EJB 3.x beans in an EJB JAR module or WAR module in `server.xml` in the `<application>` or `<ejbApplication>` elements:
+
+
+[source, XML]
+----
+   <ejbApplication location="EJBTest.jar">
+      <ejb-jar-bnd>
+         <session name="InventoryServiceBean">
+            <interface class="com.ejbs.InventoryService" binding-name="ejb/Inventory"/>
+         </session>
+      </ejb-jar-bnd>
+   </ejbApplication>
+----
+
+[customibm2]
+==== Specify Custom bindings in ibm-ejb-jar-bnd.xmi for EJB 1.x/2.x
+
+Following is an example of how to configure custom bindings for EJB 1.x or 2.x beans in an EJB JAR module in `ibm-ejb-jar-bnd.xmi`
+
+EJB 1.x and 2.x provide a single JNDI name that applies to both the remote and local home:
+
+
+[source, XML]
+----
+   <ejbBindings xmi:id="BeanBinding_8" jndiName="suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome">
+      <enterpriseBean xmi:type="ejb:Session" href="META-INF/ejb-jar.xml#SLCMTTxTimeout"/>
+   </ejbBindings>
+----
+
+For a bean with both a remote and local home, the above will provide the following custom bindings:
+
+
+[source, text]
+----
+   Remote Home : suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome
+   Local Home  : local:suite/r6x/base/misc/poollimits/SLCMTTxTimeoutHome
+----
+
+=== Feature configuration details
+
+Support for custom and legacy default JNDI names is enabled by default for all Enterprise Bean (EJB) features. This support will not interfere with the existing specification defined `java`: support. However, it is possible to completely disable the new support with the following setting in server.xml:
+
+
+[source, XML]
+----
+    <ejbContainer bindToServerRoot="false"/>
+----
+
+It is also possible to disable just the legacy short form default JNDI name support (i.e. the bean is bound using the interface name) with the following setting in server.xml:
+
+
+[source, XML]
+----
+   <ejbContainer disableShortDefaultBindings="true"/>
+----
+
+Since the new support for customer JNDI names and legacy defaults provider alternative JNDI names, it is now possible to disable the EJB specification required JNDI names. This is done in server.xml as follows:
```

Comment:
should this read ".... for **custom** JNDI names and legacy defaults **provide** alternative JNDI names...." @tkburroughs

Reply:
Done